fastdialer/dialer.go

```diff
@@ -127,10 +127,14 @@ func NewDialer(options Options) (*Dialer, error) {
 	var hostsFileData *hybrid.HybridMap
 	// load hardcoded values from host file
 	if options.HostsFile {
+		var err error
```

Comment:
I think hosts file loading should mimic the default OS behavior: if for any reason the hosts file is not accessible the system performs the DNS lookup with the specific query. What do you think of eventually accepting an optional logger similarly to `net/http`